prunning_implementation/prunning.py

```diff
@@ -0,0 +1,99 @@
+import babeltrace
+from babeltrace import CTFWriter
+import sys
+import tempfile
+from collections import Counter
+from collections import deque
+
+
+#############################################################################
+#    Function to Prunn the data and put in a deque                          #
+#    @param trace_path: The path to the stream file                #
+#           capacity: The capacity of the deque                             #
+############################################################################
+
+def getPrunnedData(trace_path, capacity):
+    
+    myDeque = deque(maxlen = capacity)
+    trace_collection = babeltrace.TraceCollection()
+    
+    # Raise an exception
+    if trace_collection.add_trace(trace_path,'ctf') is None:
+        raise RuntimeError('Cannot add trace')
+    
+    # Gather all the information and discard what you dont easy.
+    # print(trace_collection)
+    for event in trace_collection.events:
+        temp_dict = {}
+        for key in event:
+            temp_dict[key] = event[key]
+        temp_dict['event_name'] = event.name
+        if(len(myDeque) >= capacity):
+            myDeque.popleft()
+        myDeque.append(temp_dict)
```

Comment:
Why doesn't `myDeque.append(event)` just work? Why need to build this `temp_dict`? couldn't this work:
```Python
for event in trace_collection.events:
    myDequeue.append(event)
```
and let maxlen take care of dropping the events that exceed capacity?